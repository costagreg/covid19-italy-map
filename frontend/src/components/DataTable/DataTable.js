import React from 'react'
import { format } from 'date-fns'
import './DataTable.scss'

export default function DataTable({ updatesDate,  regionData, selectParam }) {
  const updatesDateFormated = format(updatesDate, 'MMMM dd yyyy')

  return (
    <div data-testid={`dataTable-${regionData['region']}`} className='dataTable'>
      <div className='dataTable__header'>
        {regionData['region']}
      </div>
      {/*TO-DO: use map for below*/}
      <div className='dataTable__row' onClick={() => selectParam('hospitalizedWithSymptoms')}>
        <div className='dataTable__col'>Hospitalized With Symptoms</div>
        <div className='dataTable__col'>{regionData['hospitalizedWithSymptoms']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col' onClick={() => selectParam('intensiveCare')}>Intensive Cares</div>
        <div className='dataTable__col'>{regionData['intensiveCare']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Total Hospitalized</div>
        <div className='dataTable__col'>{regionData['totalHospitalized']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Home Isolation</div>
        <div className='dataTable__col'>{regionData['homeIsolation']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Total Positive</div>
        <div className='dataTable__col'>{regionData['totalPositive']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Total Change Positive</div>
        <div className='dataTable__col'>{regionData['totalChangePositive']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>New Positive</div>
        <div className='dataTable__col'>{regionData['newPositive']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Discharged Healed</div>
        <div className='dataTable__col'>{regionData['dischargedHealed']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Total Deaths</div>
        <div className='dataTable__col'>{regionData['totalDeaths']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Total Cases</div>
        <div className='dataTable__col'>{regionData['totalCases']}</div>
      </div>
      <div className='dataTable__row'>
        <div className='dataTable__col'>Total Tests</div>
        <div className='dataTable__col'>{regionData['totalTests']}</div>
      </div>
      <div className='dataTable__rowDate'>The data above are update on {updatesDateFormated}</div>
    </div>
  )
}
