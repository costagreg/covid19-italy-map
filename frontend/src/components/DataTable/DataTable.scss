.dataTable{
  &__header{
    text-align: center;
  }
  &__row{
    display: flex;
    box-shadow: 0px 0px 9px 0px rgba(0,0,0,0.1);
    margin: 10px 0;
  }
  &__rowDate{
    text-align: center;
    font-size: 0.8rem;
    font-style: oblique;
  }
  &__col{
    padding: 5px 10px;
    flex:25%;

    &:first-child{
      flex: 75%;
    }
  }
}