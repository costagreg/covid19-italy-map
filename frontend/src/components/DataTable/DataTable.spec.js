import { render } from '@testing-library/react'
import React from 'react'
import DataTable from './DataTable'

const initProps = {
  regionData: {
    region: 'Sicilia',
    hospitalizedWithSymptoms: 566,
    intensiveCare: 63,
    totalHospitalized: 629,
    homeIsolation: 1313,
    totalPositive: 1942,
    totalChangePositive: 49,
    newPositive: 73,
    dischargedHealed: 152,
    totalDeaths: 138,
    totalCases: 2232,
    totalTests: 28742,
  },
}

describe('DataTable', () => {
  describe('@render', () => {
    it('renders the DataTable without any error', () => {
      const { asFragment } = render(<DataTable {...initProps} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
