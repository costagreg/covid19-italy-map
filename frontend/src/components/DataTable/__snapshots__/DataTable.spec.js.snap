// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable @render renders the DataTable  without any error 1`] = `
<DocumentFragment>
  <div
    class="dataTable"
    data-testid="dataTable-Sicilia"
  >
    <div
      class="dataTable__header"
    >
      Sicilia
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Hospitalized With Symptoms
      </div>
      <div
        class="dataTable__col"
      >
        566
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Intensive Cares
      </div>
      <div
        class="dataTable__col"
      >
        63
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Total Hospitalized
      </div>
      <div
        class="dataTable__col"
      >
        629
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Home Isolation
      </div>
      <div
        class="dataTable__col"
      >
        1313
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Total Positive
      </div>
      <div
        class="dataTable__col"
      >
        1942
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Total Change Positive
      </div>
      <div
        class="dataTable__col"
      >
        49
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        New Positive
      </div>
      <div
        class="dataTable__col"
      >
        73
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Discharged Healed
      </div>
      <div
        class="dataTable__col"
      >
        152
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Total Deaths
      </div>
      <div
        class="dataTable__col"
      >
        138
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Total Cases
      </div>
      <div
        class="dataTable__col"
      >
        2232
      </div>
    </div>
    <div
      class="dataTable__row"
    >
      <div
        class="dataTable__col"
      >
        Total Tests
      </div>
      <div
        class="dataTable__col"
      >
        28742
      </div>
    </div>
  </div>
</DocumentFragment>
`;
